contrib/nydusify/pkg/copier/copier.go

```diff
@@ -285,41 +286,78 @@ func Copy(ctx context.Context, opt Opt) error {
 	}
 	defer os.RemoveAll(tmpDir)
 
-	sourceNamed, err := docker.ParseDockerRef(opt.Source)
-	if err != nil {
-		return errors.Wrap(err, "parse source reference")
-	}
-	targetNamed, err := docker.ParseDockerRef(opt.Target)
-	if err != nil {
-		return errors.Wrap(err, "parse target reference")
-	}
-	source := sourceNamed.String()
-	target := targetNamed.String()
+	var source string
+	if strings.HasPrefix(opt.Source, "file:///") {
```

Comment:
Can we use `url.Parse` here?

Reply:
When given an absolute path, if we use `url.Parse`,  it incorrectly resolves it into `URL.host` instead of `URL.path`. So as an alternative, I've referenced [helm](https://sourcegraph.com/github.com/helm/helm/-/blob/internal/resolver/resolver.go?L256-260) and added a function getLocalPath(). A detailed description of this function is in the latest commit, PTAL, thanks!